camel/societies/workforce/workforce.py

```diff
@@ -3240,15 +3238,13 @@ async def _create_worker_node_for_task(self, task: Task) -> Worker:
 
         self._children.append(new_node)
 
-        self._notify_worker_created(
+        await self._notify_worker_created(
             new_node,
             worker_type='SingleAgentWorker',
             role=new_node_conf.role,
             metadata={'description': new_node_conf.description},
         )
-        self._child_listening_tasks.append(
-            asyncio.create_task(new_node.start())
-        )
+        self._child_listening_tasks.append(await new_node.start())
```

Comment:
```suggestion
        self._child_listening_tasks.append(asyncio.create_task(new_node.start()))
```
The start() method is an async coroutine that runs indefinitely - it's a long-running background listener that processes messages from a channel. When you write `await new_node.start()`:
  1. The execution blocks at this line
  2. It waits for start() to complete
  3. But start() never completes - it runs forever!
  4. The function never returns, so the worker is never added to the list